industrial_ci/src/docker.sh

```diff
@@ -166,7 +169,7 @@ function ici_prepare_docker_image() {
     fi
   elif [ -z "$DOCKER_IMAGE" ]; then # image was not provided, use default
      ici_build_default_docker_image
-  else
+  elif [ "$DOCKER_PULL" != false ]; then
```

Comment:
Related to https://github.com/ros-industrial/industrial_ci/pull/283/files#r201791665, it'd be friendlier if error prints what could be missing when necessary `DOCKER*` variables aren't passed. If `docker` already prints something obvious, that may be good enough though.

Reply:
I guess this follows the decision made in https://github.com/ros-industrial/industrial_ci/pull/283#discussion_r201791665